src/main/scala/com/fulcrumgenomics/bam/MakeUnmappedByInsertSize.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) $year Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import java.nio.file.Path
+import java.text.DecimalFormat
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.api.{SamRecord, SamSource, SamWriter}
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.CommonsDef.PathToBam
+import com.fulcrumgenomics.commons.io.Io
+import com.fulcrumgenomics.commons.util.LazyLogging
+import com.fulcrumgenomics.sopt._
+import htsjdk.samtools.util.IntervalList
+
+import scala.math.abs
+
+@clp(description = 
+  """
+     |Makes reads unmapped if based on their insert size.
+     |
+     |Fragment reads will always be made unmapped. Secondary and supplementary reads whose become unmapped will instead
```

Comment:
The second sentence here doesn't scan.  If I read it as "whose primary read becomes..." then it also appears false.  The code below appears, as best I can tell, to remove all secondary and supplementary reads.

Reply:
Looking at the code I'm guessing this is actually meant to be something like "reads whose insert sizes falls outside the desired range will be removed instead of being unmapped."